public/app/plugins/panel/table/module.tsx

```diff
@@ -75,10 +75,18 @@ export const plugin = new PanelPlugin<PanelOptions, PanelFieldConfig>(TablePanel
     },
   })
   .setPanelOptions((builder) => {
-    builder.addBooleanSwitch({
-      path: 'showHeader',
-      name: 'Show header',
-      description: "To display table's header or not to display",
-      defaultValue: defaultPanelOptions.showHeader,
-    });
+    builder
+      .addBooleanSwitch({
+        path: 'showHeader',
+        name: 'Show header',
+        description: "To display table's header or not to display",
+        defaultValue: defaultPanelOptions.showHeader,
+      })
+      .addBooleanSwitch({
+        path: 'showTypeIcons',
+        name: 'Show type icons',
+        description: 'Include an icon indicating the field type',
+        defaultValue: defaultPanelOptions.showTypeIcons,
+        showIf: (ctx) => !!ctx.showHeader,
+      });
```

Comment:
@ryantxu @torkelo only now I've noticed that this information is shown in the Table view, not the Table panel. If so - this sounds better to me. But in that case, since the Table view is used to preview the data, I would expect the boolean switch not to be Table panel option. Or at least hidden from the options that are available via UI.

Reply:
yea, agree that it should not be a UI option for the table panel itself

Reply:
removed